aws/resource_aws_sso_permission_set.go

```diff
@@ -0,0 +1,642 @@
+		if permissionerr != nil {
+			return fmt.Errorf("Error updating AWS SSO Permission Set: %s", permissionerr)
+		}
+	}
+
+	if d.HasChange("tags") {
+		oldTags, newTags := d.GetChange("tags")
+		if updateTagsErr := keyvaluetags.SsoUpdateTags(ssoadminconn, d.Get("arn").(string), d.Get("instance_arn").(string), oldTags, newTags); updateTagsErr != nil {
+			return fmt.Errorf("Error updating tags: %s", updateTagsErr)
+		}
+	}
+
+	if v, ok := d.GetOk("inline_policy"); ok {
+		log.Printf("[DEBUG] AWS SSO Permission Set %s updating IAM inline policy", permissionSetArn)
+
+		inlinePolicy := aws.String(v.(string))
+
+		updateInput := &ssoadmin.PutInlinePolicyToPermissionSetInput{
+			InlinePolicy:     inlinePolicy,
+			InstanceArn:      aws.String(instanceArn),
+			PermissionSetArn: aws.String(permissionSetArn),
+		}
+
+		_, inlinePolicyErr := ssoadminconn.PutInlinePolicyToPermissionSet(updateInput)
+		if inlinePolicyErr != nil {
+			return fmt.Errorf("Error attaching IAM inline policy to AWS SSO Permission Set: %s", inlinePolicyErr)
+		}
+	} else if d.HasChange("inline_policy") {
+		deleteInput := &ssoadmin.DeleteInlinePolicyFromPermissionSetInput{
+			InstanceArn:      aws.String(instanceArn),
+			PermissionSetArn: aws.String(permissionSetArn),
+		}
+
+		_, inlinePolicyErr := ssoadminconn.DeleteInlinePolicyFromPermissionSet(deleteInput)
+		if inlinePolicyErr != nil {
+			return fmt.Errorf("Error deleting IAM inline policy from AWS SSO Permission Set: %s", inlinePolicyErr)
+		}
+	}
+
+	if d.HasChange("managed_policy_arns") {
+		o, n := d.GetChange("managed_policy_arns")
+
+		os := o.(*schema.Set)
+		ns := n.(*schema.Set)
+
+		removalList := os.Difference(ns)
+		for _, v := range removalList.List() {
+			input := &ssoadmin.DetachManagedPolicyFromPermissionSetInput{
+				InstanceArn:      aws.String(instanceArn),
+				ManagedPolicyArn: aws.String(v.(string)),
+				PermissionSetArn: aws.String(permissionSetArn),
+			}
+
+			_, managedPoliciesErr := ssoadminconn.DetachManagedPolicyFromPermissionSet(input)
+			if managedPoliciesErr != nil {
+				return fmt.Errorf("Error detaching Managed Policy from AWS SSO Permission Set: %s", managedPoliciesErr)
+			}
+		}
+
+		additionList := ns.Difference(os)
+		for _, v := range additionList.List() {
+			input := &ssoadmin.AttachManagedPolicyToPermissionSetInput{
+				InstanceArn:      aws.String(instanceArn),
+				ManagedPolicyArn: aws.String(v.(string)),
+				PermissionSetArn: aws.String(permissionSetArn),
+			}
+
+			_, managedPoliciesErr := ssoadminconn.AttachManagedPolicyToPermissionSet(input)
+			if managedPoliciesErr != nil {
+				return fmt.Errorf("Error attaching Managed Policy to AWS SSO Permission Set: %s", managedPoliciesErr)
+			}
+		}
+	}
+
+	// Reprovision if anything has changed
+	if d.HasChanges("description", "relay_state", "session_duration", "inline_policy", "managed_policy_arns", "tags") {
+
+		// Auto provision all accounts
+		targetType := ssoadmin.ProvisionTargetTypeAllProvisionedAccounts
```

Comment:
Iooking at how cloudformation doe this, I think provisioning to `ALL` accounts makes sense to make the uniform change, I would just note this in the documentation as it occurs behind the scenes